minimint-core/src/transaction.rs

```diff
@@ -63,7 +63,7 @@ impl TransactionItem for Output {
     fn amount(&self) -> Amount {
         match self {
             Output::Mint(coins) => coins.amount(),
-            Output::Wallet(peg_out) => peg_out.amount.into(),
+            Output::Wallet(peg_out) => (peg_out.amount + peg_out.fees.amount()).into(),
```

Comment:
Unrelated to this PR, but I feel like we should get rid of this API and only rely on what the modules return from `apply_*`

Reply:
Guess you are thinking modularly now :)

Reply:
After a day of haskell-like rust definitely :laughing: